### PR TITLE
Use actual equivalence keys, not loc strings to distinguish helpers

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/62739

### CODE REVIEW

File: src/Analyzers/CSharp/Analyzers/UseExpressionBody/Helpers/UseExpressionBodyHelper.cs [hunk 1]
```diff
@@ -14,8 +14,13 @@ namespace Microsoft.CodeAnalysis.CSharp.UseExpressionBody
     internal abstract class UseExpressionBodyHelper
     {
         public abstract Option2<CodeStyleOption2<ExpressionBodyPreference>> Option { get; }
+
         public abstract LocalizableString UseExpressionBodyTitle { get; }
         public abstract LocalizableString UseBlockBodyTitle { get; }
+
+        public abstract string UseExpressionBodyEquivalenceKey { get; }
```
Comment (tmat) [inline src/Analyzers/CSharp/Analyzers/UseExpressionBody/Helpers/UseExpressionBodyHelper.cs:21]: >UseExpressionBodyEquivalenceKey [](http://example.com/codeflow?start=31&length=31)

Couldn't these be `GetType().Name`?

File: src/Analyzers/CSharp/Analyzers/UseExpressionBody/Helpers/UseExpressionBodyHelper.cs [hunk 2]
```diff
@@ -14,8 +14,10 @@ namespace Microsoft.CodeAnalysis.CSharp.UseExpressionBody
     internal abstract class UseExpressionBodyHelper
     {
         public abstract Option2<CodeStyleOption2<ExpressionBodyPreference>> Option { get; }
+
         public abstract LocalizableString UseExpressionBodyTitle { get; }
         public abstract LocalizableString UseBlockBodyTitle { get; }
+
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/Analyzers/UseExpressionBody/Helpers/UseExpressionBodyHelper.cs:20]:
```suggestion
        public abstract LocalizableString UseExpressionBodyTitle { get; }
        public abstract LocalizableString UseBlockBodyTitle { get; }
```

**Review by tmat — APPROVED** [review]
:shipit:

File: src/Features/CSharp/Portable/UseExpressionBody/UseExpressionBodyCodeRefactoringProvider.cs [hunk 3]
```diff
@@ -125,28 +124,26 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
...
-                var title = s_helperToTitleMap[(helper, useExpressionBody: false)];
                 context.RegisterRefactoring(
                     CodeAction.Create(
-                        title,
+                        helper.UseBlockBodyTitle.ToString(),
                         c => UpdateDocumentAsync(
                             document, root, declaration, helper,
                             useExpressionBody: false, cancellationToken: c),
-                        title),
+                        s_equivalenceKeyMap[(helper, useExpressionBody: false)]),
```
Comment (mavasani) [inline src/Features/CSharp/Portable/UseExpressionBody/UseExpressionBodyCodeRefactoringProvider.cs:146]: I think such a pattern is used by many analyzers. @Youssef1313 did clean up bunch of these, but quite a few still exist in our code base.

**Review by mavasani — APPROVED** [review]

**Pull request merged**